otx/algorithms/classification/tasks/inference.py

```diff
@@ -351,11 +351,18 @@ def _init_recipe(self):
 
         if train_type == TrainType.INCREMENTAL:
             recipe = os.path.join(recipe_root, "incremental.yaml")
+            if self._hyperparams.learning_parameters.enable_supcon:
+                recipe = os.path.join(recipe_root, "supcon.yaml")
+                self.task_pipeline_path = os.path.join(
```

Comment:
I am not sure I follow, I did it like in the selfsl case.

Reply:
I didn't update the `self.model_dir` to contain `supcon` because it won't work when the task is `multilabel` or `hierarchical`. In my next PR, which adds support for those two tasks, I will do it.

Reply:
yes

Reply:
Then why supcon generate new variable task_model_dir and task_pipeline_path?

Reply:
I don't understand your question. Supcon needs its own model and data pipeline.

Reply:
My point is why we should change variable name from self.model_dir to self.task_model_dir and from self.pipeline_path to self.task_pipeline_path.
Current feature/otx branch already merged Sunchul's cls-selfsl PR and he agree the variable names self.model_dir and self.pipeline_path

Reply:
Ah, I see it now, it was changed after I merged the first time, I will fix that. I didn't make this names up, they were carried out from a previous version. Sorry for the confusion.

Reply:
No problem!